File: Prácticas/P5/JAVA/Civitas/src/civitas/CasillaJuez.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package civitas;

import java.util.ArrayList;

/**
 *
 * @author Jose Padial
 */
public class CasillaJuez extends Casilla{
    private static int carcel;
    
    CasillaJuez(String nombre, int numCasillaCarcel){
        super(nombre);
        CasillaJuez.carcel = numCasillaCarcel;
    }
    
    @Override
    void recibeJugador(int actual, ArrayList<Jugador> todos){
        if(jugadorCorrecto(actual,todos) == true){
            informe(actual,todos);
            todos.get(actual).encarcelar(carcel);
        }
    }

    @Override
    public String toString() {
        return "CasillaJuez{" + "Numero=" + carcel + '}';
    }
}

File: Prácticas/P4/RUBY/Civitas/lib/casilla_juez.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.

module Civitas
  class CasillaJuez < Casilla
    def initialize(nombre,num_casilla_carcel)
      @@carcel = num_casilla_carcel
      super(nombre)
    end
    
    def recibe_jugador(actual,todos)
      if(jugador_correcto(actual,todos))
        informe(actual, todos)
        todos[actual].encarcelar(@@carcel)
      end
    end
    
    def to_s
      to_s="CasillaJuez carcel= #{@@carcel}"
    end
  end
end
